restaurants/models.py

```diff
@@ -0,0 +1,263 @@
+    owner_email(EmailField): Email address of restaurant owner
+    license_registration_image(FileField): FSSAI license image
+    license_registration_number(IntegerField): FSSAI license number
+    license_expiration_date(DateField): FSSAI license expire date
+    bank_account_image(ManyToManyField): Image of bank passbook / cancel cheque
+    bank_account_number(CharField): Bank account number
+    bank_ifsc_code(CharField): Bank IFSC code
+    parent_restaurant(ForeignKey): self referencing foreign key to ensure about restaurant's multiple branches.
+    location(ForeignKey): Location of restaurant
+    manager(ForeignKey): Manager will be added in CustomUser model as a type = 'manager' after restaurant request will approved.
+
+    """
+
+    OUTLET = (
+        ("Corner Cafe", "Corner Cafe"),
+        ("Main Street Diner", "Main Street Diner"),
+        ("SunnySide Grill", "SunnySide Grill"),
+        ("RiverSide Bistro", "RiverSide Bistro"),
+    )
+    name = models.CharField(max_length=255)
+    phone = models.IntegerField()
+    outlet = models.CharField(max_length=255, choices=OUTLET)
+    invoicing_email = models.EmailField()
+    is_open = models.BooleanField(default=False)
+    owner_name = models.CharField(max_length=255)
+    owner_email = models.EmailField()
+
+    license_registration_image = models.FileField()
+    license_registration_number = models.IntegerField()
+    license_expiration_date = models.DateField()
+
+    bank_account_image = models.ManyToManyField(Image)
+    bank_account_number = models.CharField(max_length=255)
+    bank_ifsc_code = models.CharField(max_length=11)
+
+    parent_restaurant = models.ForeignKey(
+        "self",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="branches",
+    )
+    location = models.ForeignKey(
+        Location,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="restaurants",
+    )
+    manager = models.ForeignKey(
+        CustomUser,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="restaurants",
+    )
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        db_table = "restaurants_restaurant"
+
+
+class RestaurantDocument(BaseModel):
+    """
+    This model represents restaurant documents.
+
+    Attributes:
+
+    restaurant(ForeignKey): Represents restaurant id
+    document_type(Enum): Type of document
+    document_number(CharField): Document number
+    document_image(FileFIeld): Image of document
+    """
+
+    STATUS = (
+        ("PAN", "PAN"),
+        ("GST", "GST"),
+    )
+    restaurant = models.ForeignKey(
+        Restaurant, on_delete=models.CASCADE, related_name="restaurantdocument"
+    )
+    document_type = models.CharField(max_length=255, choices=STATUS)
+    document_number = models.CharField(max_length=255)
+
+    document_image = models.FileField(upload_to="Documents/")
```

Comment:
Create a function for uploading doc to specific folder. Take help of Ravindra bhai, as he already did this in DDatori as well.